apps/nextjs/src/components/DialogControl/ContentOptions/ClearSingleChatFromChatHistory.tsx

```diff
@@ -0,0 +1,85 @@
+"use client";
+
+import { useTransition } from "react";
+import { toast } from "react-hot-toast";
+
+import {
+  OakFlex,
+  OakP,
+  OakSmallPrimaryButton,
+  OakSmallSecondaryButton,
+} from "@oaknational/oak-components";
+import { showReportDialog } from "@sentry/nextjs/build/types/server";
+
+import LoadingWheel from "@/components/LoadingWheel";
+import { trpc } from "@/utils/trpc";
+
+const ClearSingleChatFromChatHistory = ({
+  closeDialog,
+}: {
+  closeDialog: () => void;
+}) => {
+  const { mutate, isLoading } =
+    trpc.chat.appSessions.deleteChatById.useMutation({
+      onSuccess() {
+        console.log("Chat history cleared");
+        closeDialog();
+        localStorage.removeItem("chatIdToDelete");
+      },
+      onError() {
+        console.error("Failed to clear chat history");
+      },
+    });
+
+  function deleteChatById() {
+    const id = localStorage.getItem("chatIdToDelete");
```

Comment:
This really shows me that the way we're doing dialogs isn't appropriate at all. Maybe you could set it on the dialog context?

We don't really need localstorage for this as it doesn't need to persist. You could equally set a value on `window` on on an object exported from another file

If we think we'll tidy this up later then we could go ahead with this knowing that it's not the right way to be doing it

Reply:
Localstorage is sometimes not available on some school computers so this might throw and make it so you can't clear the history

Reply:
Yeah we have ran into an issue here. 
Traditionally, with this set up we can pass props to a dialog directly however due to the knarly css here the dialog nested at this point does not work at all. I can have another look now im zoomed out a bit and see if I can refactor the css to allow for the dialog to sit at this level. Alt a some context could work too